addons/stock/models/stock_picking.py

```diff
@@ -670,7 +671,9 @@ def _compute_state(self):
                     picking.state = 'done'
             else:
                 relevant_move_state = self.env['stock.move'].browse(picking_move_lines[picking_id])._get_relevant_state_among_moves()
-                if relevant_move_state == 'partially_available':
+                if picking.picking_type_code == 'incoming' and picking.location_id.should_bypass_reservation():
```

Comment:
```suggestion
                if picking.location_id.should_bypass_reservation():
```

Reply:
and we could call it before relevant_move_state = ... to avoid useless computation